sklearn/datasets/samples_generator.py

```diff
@@ -162,17 +162,18 @@ def make_classification(n_samples=100, n_features=20, n_informative=2,
     if n_informative < np.log2(n_classes * n_clusters_per_class):
         raise ValueError("n_classes * n_clusters_per_class must"
                          " be smaller or equal 2 ** n_informative")
-    if weights and len(weights) not in [n_classes, n_classes - 1]:
-        raise ValueError("Weights specified but incompatible with number "
+    if not weights is None:
```

Comment:
the idiomatic python way is ``weights is not None``.
I would use a single ``if`` and ``and``. You can put the whole condition in parenthesis and make it multi-line

Reply:
ok.